docs/proposals/workflow.md

```diff
@@ -0,0 +1,134 @@
+A similar approach is implemented in Chronos for [dependent jobs](https://mesos.github.io/chronos/docs/api.html#adding-a-dependent-job).
+
+### API Object
+
+To implement _workflows_ the `Job` API should be modified:
+
+```go
+// JobsSelector it's just an alias for PodSelector
+type JobSelector PodSelector
+
+// Job represents the configuration of a single job.
+type Job struct {
+    unversioned.TypeMeta `json:",inline"`
+    // Standard object's metadata.
+
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+    api.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec is a structure defining the expected behavior of a job.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+    Spec JobSpec `json:"spec,omitempty"`
+
+    // Status is a structure describing current status of a job.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+    Status JobStatus `json:"status,omitempty"`
+
+    // JobSelector to detect parents jobs
+    ParentSelector *JobSelector `json:"parentSelector, omitempty"`
+}
+```
+
+#### JobSpec and JobStatus
+
+No modifications
+
+#### Parent JobSelector
+
+The `.job.parentselector` is a label query over a set of jobs.
+If all selected jobs are completed the `job` will be started immediately.
+If `.job.parentselector` is absent the `job` will be started immediately.
+Since labels are forwarded directly from `job` to pods, users should not create pods whose labels match this selector, either directly,
+via another Job, or via another controller (for example Replication controller), [see](https://github.com/kubernetes/kubernetes/issues/14961).
+
+#### Some special cases
+
+Jobs terminated via [#17244](https://github.com/kubernetes/kubernetes/issues/17244)  (if implemented) will be considered _completed_ so will trigger the _child_ jobs. Since it's very unlikely that the user would keep the _child_ jobs alive we may add a field to cascade termination to _child_ jobs if any to the current proposals made by @pmorie and @bgrant0607.
+
+For example
+
+```json
+apiVersion: v1alpha1
+kind: UpwardAPIRequest
+spec:
+    terminateExistingJob: true
+    reason: "TimeOut"
+    message: "Pods ran out of time"
+    cascade: true
+```
+
+### CRUD
+
+    - Creating a job without an empty `jobs.parentselector` will simply follow the usual `job` life cycle.  With a non non nil `jobs.parentselector` the logic already described will be considered.
```

Comment:
I'd create two separate points, one describing non-empty parent selector, and the other with empty.